addons/survey/controllers/main.py

```diff
@@ -726,39 +721,101 @@ def _get_user_input_domain(self, survey, line_filter_domain, **post):
+                    answer_domain = ['&', '&', ('matrix_row_id', '=', row_id), ('suggested_answer_id', '=', answer_id), ('question_id', '=', question.id)]
+                else:
+                    answer = request.env['survey.user_input.line'].browse([answer_id])
+                    question = answer.question_id
+
+                    if question_type == 'text_box':
+                        value = answer.value_text_box
+                        answer_domain = ['&', ('value_text_box', 'ilike', value), ('question_id', '=', question.id)]
+                    elif question_type == 'char_box':
+                        value = answer.value_char_box
+                        answer_domain = ['&', ('value_char_box', '=ilike', value), ('question_id', '=', question.id)]
+                    elif question_type == 'numerical_box':
+                        value = answer.value_numerical_box
+                        answer_domain = ['&', ('value_numerical_box', '=', value), ('question_id', '=', question.id)]
+                    elif question_type == 'date':
+                        value = answer.value_date
+                        answer_domain = ['&', ('value_date', '=', value), ('question_id', '=', question.id)]
+                    elif question_type == 'datetime':
+                        value = answer.value_datetime
+                        answer_domain = ['&', ('value_datetime', '=', value), ('question_id', '=', question.id)]
+
+                if question_type in ('simple_choice', 'multiple_choice', 'matrix'):
+                    value = '%s%s' % (answer[0].value, ': %s' % answer[1].value if len(answer) > 1 else '')
+
+                if answer_domain:
+                    filters_domain.append(answer_domain)
+                filters_data.append({
+                    'question_type': question_type,
+                    'row_id': row_id,
+                    'answer_id': answer_id,
+                    'question_id': question.id,
+                    'question': question.title,
+                    'answers': value
+                })
+
+        # Get the survey user inputs
+        user_input_domain = self._get_user_input_domain(survey, **post)
+        user_inputs = request.env['survey.user_input'].sudo().search(user_input_domain)
+
+        # Create the domain to collect all the user_input_lines matching the filters independently
+        input_line_domain = ['&', ('user_input_id', 'in', user_inputs.ids), ('suggested_answer_id', 'in', suggested_answer_ids)]
+        for domain in filters_domain:
+            input_line_domain = expression.OR([input_line_domain, domain])
+        results = request.env['survey.user_input.line'].search_read(
+            input_line_domain,
+            fields=['id', 'user_input_id', 'question_id', 'answer_type', 'suggested_answer_id', 'value_char_box', 'matrix_row_id']
+        )
+
+        # Then, keep only the user input ids that match all the filters
+        user_input_by_answer_id = defaultdict(list)
```

Comment:
Not sure I understand this part.

Seems like the previous logic was:
- find all lines matching our criteria
- find user inputs for these lines
- make a AND between the base upser inputs filter and these lines
- return ALL lines for these inputs

I see we changed it here, but not sure why, was it not working?